webhooks-extension/pkg/endpoints/webhook_test.go

```diff
@@ -67,1024 +103,632 @@ func TestGitHubSource(t *testing.T) {
+	var hooks = []webhook{
+		{
+			Name:             "name1",
+			Namespace:        installNs,
+			GitRepositoryURL: "https://github.com/owner/repo",
+			AccessTokenRef:   "token1",
+			Pipeline:         "pipeline1",
+			DockerRegistry:   "registry1",
+			HelmSecret:       "helmsecret1",
+			ReleaseName:      "releasename1",
+			PullTask:         "pulltask1",
+			OnSuccessComment: "onsuccesscomment1",
+			OnFailureComment: "onfailurecomment1",
+		},
+		{
+			Name:             "name2",
+			Namespace:        "foo",
+			GitRepositoryURL: "https://github.com/owner/repo",
+			AccessTokenRef:   "token2",
+			Pipeline:         "pipeline2",
+			DockerRegistry:   "registry2",
+			OnSuccessComment: "onsuccesscomment2",
+			OnFailureComment: "onfailurecomment2",
+		},
 	}
 
-	// Now compare the arrays expectedWebhooks and actualWebhooks by turning them into maps
-	expected := map[webhook]bool{}
-	actual := map[webhook]bool{}
-	for i := range expectedWebhooks {
-		if expectedWebhooks[i].DockerRegistry == "" {
-			expectedWebhooks[i].DockerRegistry = defaultRegistry
-		}
-		expected[expectedWebhooks[i]] = true
-		actual[actualWebhooks[i]] = true
+	r := dummyResource()
+	os.Setenv("SERVICE_ACCOUNT", "tekton-test-service-account")
+	el, err := r.createEventListener(hooks[0], "install-namespace", hooks[0].GitRepositoryURL[strings.LastIndex(hooks[0].GitRepositoryURL, ":")+3:])
+	if err != nil {
+		t.Errorf("Error creating eventlistener: %s", err)
 	}
-
-	if !reflect.DeepEqual(expected, actual) {
-		t.Errorf("Webhook error: expected: \n%v \nbut received \n%v", expected, actual)
+	el, err = r.updateEventListener(el, hooks[1], hooks[1].GitRepositoryURL[strings.LastIndex(hooks[1].GitRepositoryURL, ":")+3:])
+	if err != nil {
+		t.Errorf("Error updating eventlistener: %s", err)
 	}
-}
 
-// Checks that URLs without .git match a .git URL in the configmap
-func testGetGitHubWebhook(gitURL string, r *Resource, t *testing.T) {
-
-	configMapClient := r.K8sClient.CoreV1().ConfigMaps(installNs)
-	_, err := configMapClient.Get(ConfigMapName, metav1.GetOptions{})
-	if err != nil {
-		t.Errorf("Uh oh, we got an error looking up the ConfigMap for webhooks. Error is: %s", err)
+	if len(el.Spec.Triggers) != 5 {
+		t.Errorf("Eventlistener had %d triggers, but expected 5", len(el.Spec.Triggers))
 	}
 
-	if strings.HasSuffix(gitURL, ".git") {
-		blank := webhook{}
-		// Trim the suffix - as this is how it comes from the actual webhook
-		// r.getGitHubWebhooks should still match and return the webhook from the configmap
-		whooksReturned, err := r.getGitHubWebhooks(strings.TrimSuffix(gitURL, ".git"))
-		if err != nil {
-			t.Errorf("Error occurred in getGitHubWebhooks: %s", err)
-		} else {
-			if len(whooksReturned) == 0 || len(whooksReturned) > 1 {
-				t.Errorf("An incorrect number of webhooks were returned from getGitHubWebhooks")
-				t.Fail()
+	expectedTriggers := []v1alpha1.EventListenerTrigger{}
+	triggerNamesExpected := make(map[string]string)
```

Comment:
This is the second 'triggerNamesExpected' method deserving a comment and possible consolidation, as per the note above.